src/Common/Core/Impl/Extensions/CultureExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace Microsoft.Common.Core {
+    public static class CultureExtensions {
+        private static readonly char[] _separators = new char[] { ',', ' ' };
+
+        public static string LanguageNameFromLCID(int lcid) {
+            var culture = CultureInfo.GetCultures(CultureTypes.InstalledWin32Cultures).FirstOrDefault(x => x.LCID == lcid);
+            return culture.GetLanguageName();
+        }
+
+        public static IEnumerable<string> GetLanguageNames() {
+            IEnumerable<CultureInfo> cultures = CultureInfo.GetCultures(CultureTypes.InstalledWin32Cultures);
+            return cultures.Select(c => c.GetLanguageName()).Distinct().OrderBy(x => x);
```

Comment:
It is indeed an interesting problem. There is no really a good list of what particular C runtime accepts in LC_CTYPE. I can ping MSVCRT folks and get their list. .NET list does not map 1:1 and there is no 'country' name there. RS only provides ~10 encodings in their list although Sys.setlocale accepts quite a few more. I can try and make list like Language.LCID if the helps?